xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java

```diff
@@ -325,15 +325,14 @@ private void updateSslContextProvider(UpstreamTlsContext newUpstreamTlsContext)
     @Override
     public void onResourceDoesNotExist(String resourceName) {
       logger.log(XdsLogLevel.INFO, "Resource {0} is unavailable", resourceName);
-      // TODO(chengyuanzhang): should unconditionally propagate to downstream instances and
-      //  go to TRANSIENT_FAILURE.
-      if (edsBalancer == null) {
-        helper.updateBalancingState(
-            TRANSIENT_FAILURE,
-            new ErrorPicker(
-                Status.UNAVAILABLE.withDescription(
-                    "Resource " + resourceName + " is unavailable")));
+      if (edsBalancer != null) {
+        edsBalancer.shutdown();
+        edsBalancer = null;
```

Comment:
Also set `edsBalancer = null` at https://github.com/grpc/grpc-java/pull/7081/files#diff-78c989ecc725727447eb9aaf38e31055R182 to avoid `edsBalancer` double shutdown.

Reply:
It will not be the case per offline discussion.